src/renderer/components/kube-object-menu/kube-object-menu.test.tsx

```diff
@@ -0,0 +1,295 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import React from "react";
+import { render } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import { KubeObjectMenu, KubeObjectMenuDependencies } from "./kube-object-menu";
+import { KubeObject } from "../../../common/k8s-api/kube-object";
+import userEvent from "@testing-library/user-event";
+import type { KubeApi } from "../../../common/k8s-api/kube-api";
+import type { KubeObjectMenuRegistration } from "../../../extensions/registries";
+import type { IGettableStore } from "../../../common/k8s-api/api-manager";
+import type { KubeObjectStore } from "../../../common/k8s-api/kube-object.store";
+import { ConfirmDialog } from "../confirm-dialog";
+import asyncFn from "@async-fn/jest";
+import { KubeObjectMenuRegistry } from "../../../extensions/registries";
+
+describe("kube-object-menu", () => {
+  let hideDetailsStub: () => void;
+  let editResourceTabStub: () => void;
+  let apiManagerStub: IGettableStore;
+  let kubeObjectMenuRegistry: KubeObjectMenuRegistry;
+  let objectStub: KubeObject | null;
+  let dependencies: KubeObjectMenuDependencies<KubeObject>;
+
+  beforeEach(() => {
+    // TODO: Remove illegal global overwrites for what should be a dependency somewhere.
+    // TODO: Remove usage of experimental browser API.
+    window.requestIdleCallback = (callback: IdleRequestCallback): number => {
```

Comment:
Do you need to reset this in `afterAll` or `afterEach`?

Reply:
Good question. In perfect world we shouldn't need to do that but not sure at the moment. So this is something that we need to verify.